packages/jest-emotion/test/react-enzyme.test.js

```diff
@@ -1,130 +1,201 @@
+            </p>
+          }
+        >
+          World!
+        </Greeting>
+      )
+    }
+  },
+  theming: {
+    render() {
+      const Button = styled.button`
+        color: ${props => props.theme.main};
+        border: 2px solid ${props => props.theme.main};
+      `
 
-  const tree = enzyme.mount(
-    <div>
-      <Greeting
-        content={
-          <p id="something" css={{ backgroundColor: 'blue' }}>
-            Hello
-          </p>
+      Button.defaultProps = {
+        theme: {
+          main: 'red'
         }
-      >
-        World!
-      </Greeting>
-    </div>
-  )
-  expect(tree).toMatchShallowSnapshot()
-})
-
-test('enzyme test with prop containing css element with other props', () => {
-  const Greeting = ({ children, content }) => (
-    <div>
-      {content} {children}
-    </div>
-  )
+      }
 
-  const tree = enzyme.mount(
-    <Greeting
-      content={
-        <p id="something" css={{ backgroundColor: 'blue' }}>
-          Hello
-        </p>
+      const theme = {
+        main: 'blue'
       }
-    >
-      World!
-    </Greeting>
-  )
-  expect(tree).toMatchShallowSnapshot()
-})
 
-test('enzyme test with prop containing css element with other label', () => {
-  const Thing = ({ content, children }) => {
-    return children
+      return (
+        <div>
+          <Button>Normal</Button>
+          <ThemeProvider theme={theme}>
+            <Button>Themed</Button>
+          </ThemeProvider>
+        </div>
+      )
+    }
   }
-  const Greeting = ({ children, content }) => (
-    <Thing content={<div css={{ color: 'hotpink' }} />}>
-      {content} {children}
-    </Thing>
-  )
+}
 
-  const tree = enzyme.mount(
-    <Greeting
-      content={
-        <p id="something" css={{ backgroundColor: 'blue' }}>
-          Hello
-        </p>
-      }
-    >
-      World!
-    </Greeting>
+describe('enzyme', () => {
+  jestInCase(
+    'shallow',
+    ({ render }) => {
+      const wrapper = enzyme.shallow(render())
+      expect(wrapper).toMatchSnapshot()
+    },
+    cases
   )
-  expect(tree).toMatchShallowSnapshot()
-})
 
-test('enzyme test with prop containing css element in fragment', () => {
-  const FragmentComponent = () => (
-    <React.Fragment>
-      x<div css={{ backgroundColor: 'blue' }}>y</div>
-    </React.Fragment>
+  jestInCase(
+    'mount',
+    ({ render }) => {
+      const wrapper = enzyme.mount(render())
+      expect(wrapper).toMatchSnapshot()
+    },
+    cases
   )
 
-  const tree = enzyme.mount(
-    <div>
-      <FragmentComponent />
-    </div>
-  )
-  expect(tree).toMatchDeepSnapshot()
+  test('with prop containing css element in fragment', () => {
+    const FragmentComponent = () => (
+      <React.Fragment>
+        x<div css={{ backgroundColor: 'blue' }}>y</div>
+      </React.Fragment>
+    )
+
+    const wrapper = enzyme.mount(
+      <div>
+        <FragmentComponent />
+      </div>
+    )
+
+    expect(toJson(wrapper, { mode: 'deep' })).toMatchSnapshot()
```

Comment:
Is this toJson call meant to be here, I thought that's happening at the snapshot serializer level?

Reply:
this particular test requires the `deep` option, which enforces all the nodes to have their render methods called.